Demo/Demo/ViewModels/PayPalWebViewModel.swift

```diff
@@ -70,83 +72,44 @@ class PayPalWebViewModel: ObservableObject, PayPalWebCheckoutDelegate {
         }
     }
 
-    func paypalWebCheckoutSuccessResult(checkoutResult: PayPalWebState.CheckoutResult) {
-        DispatchQueue.main.async {
-            self.state.checkoutResultResponse = .loaded(checkoutResult)
-        }
-    }
-
-    func paypalWebCheckoutFailureResult(checkoutError: CorePayments.CoreSDKError) {
-        DispatchQueue.main.async {
-            self.state.checkoutResultResponse = .error(message: checkoutError.localizedDescription)
-        }
-    }
-
-    func completeOrder(with intent: Intent, orderID: String, selectedMerchantIntegration: MerchantIntegration) async throws {
-        switch intent {
-        case .capture:
-            try await captureOrder(orderID: orderID, selectedMerchantIntegration: selectedMerchantIntegration)
-            print("Order Captured. ID: \(state.capturedOrder?.id ?? "")")
-        case .authorize:
-            try await authorizeOrder(orderID: orderID, selectedMerchantIntegration: selectedMerchantIntegration)
-            print("Order Authorized. ID: \(state.authorizedOrder?.id ?? "")")
-        }
-    }
-
-    func captureOrder(orderID: String, selectedMerchantIntegration: MerchantIntegration) async throws {
+    func completeTransaction() async throws {
         do {
             DispatchQueue.main.async {
-                self.state.capturedOrderResponse = .loading
+                self.state = .loading
             }
-            let order = try await DemoMerchantAPI.sharedService.captureOrder(
-                orderID: orderID,
-                selectedMerchantIntegration: selectedMerchantIntegration
-            )
-            DispatchQueue.main.async {
-                self.state.capturedOrderResponse = .loaded(order)
-            }
-        } catch {
-            DispatchQueue.main.async {
-                self.state.capturedOrderResponse = .error(message: error.localizedDescription)
-            }
-            print("Error capturing order: \(error.localizedDescription)")
-        }
-    }
 
-    func authorizeOrder(orderID: String, selectedMerchantIntegration: MerchantIntegration) async throws {
-        do {
-            DispatchQueue.main.async {
-                self.state.authorizedOrderResponse = .loading
-            }
-            let order = try await DemoMerchantAPI.sharedService.authorizeOrder(
-                orderID: orderID,
-                selectedMerchantIntegration: selectedMerchantIntegration
-            )
-            DispatchQueue.main.async {
-                self.state.authorizedOrderResponse = .loaded(order)
+            if let orderID = order?.id {
+                let order = try await DemoMerchantAPI.sharedService.completeOrder(intent: intent, orderID: orderID)
+                
+                DispatchQueue.main.async {
+                    self.order = order
+                    self.state = .success
+                }
```

Comment:
random: I wonder if one day we'll be able to remove calls to `DispatchQueue.main.async`. If we can find a cleaner way to update the UI without having to explicitly dispatch to the main thread, we can DRY up the code and make it more readable.

Reply:
I believe the work with `@MainActor` would handle this as it would call these updates from the main thread always. Specifically we can make the entire view model a `MainActor` and remove all calls to `DispatchQueue.main` in this class. Makes sense to leave for now but that workstream should greatly clean up all the main queue calls here.

Reply:
Nice yeah that's sort of what I was thinking. It's out of scope for this PR but it'd be nice to establish that pattern one day to clean everything up.

Reply:
I did move these updates into their own methods here to keep the calls to the dispatch queue in 1 place, which cleans it up a bit more for now: https://github.com/paypal/paypal-ios/pull/223/commits/a7b9b64a46f2d4e2c29f8950f676fa806711ab6f